### ISSUE TITLE
Incorporate the timeout examples gist into specs2

### ISSUE THREAD

**Issue by ittaiz** [etorreborre/specs2#322]
Hi,
I just google'd how to timeout an example (like JUnit timeout on @Test) and saw your [gist] (https://gist.github.com/etorreborre/7885638).
My question is why not incorporate it into the library?
It's pretty obvious people want it since you created the gist from people's requests (also surfaced in the specs2 google group) and now people have to copy paste this gist into their code causing code duplication and handling something which is, IMO, more in the library scope then in the user's scope.

I'm mainly talking about the "upTo" method BTW

**Comment by etorreborre** [comment]
I have actually added it to the next [specs2 major version](https://github.com/etorreborre/specs2/blob/specs2-three/matcher-extra/src/main/scala/org/specs2/specification/ExamplesTimeout.scala).

There will also be some control over the execution model which already has a notion of timeout built-in, so you can write:

```scala
class TestSpec extends Specification { def is = s2"""
 test $e1
"""

  def e1 = Execution.result {
    Thread.sleep(2000)
    ok
  }.setTimeout(1.seconds)
}
```

**Comment by ittaiz** [comment]
Excellent, thanks.
Any chance for a back port of just the trait?

‏בתאריך יום חמישי, 22 בינואר 2015, Eric Torreborre <notifications@github.com>
כתב:

**Comment by etorreborre** [comment]
Just a matter of time... I'll try to do that next week (moving houses this week).

**Comment by etorreborre** [comment]
This is published as 2.4.16-SNAPSHOT.

**Comment by ittaiz** [comment]
You rock, thanks!
